testsuite/pytests/test_parameter_operators.py

```diff
@@ -94,37 +96,32 @@ def test_binary_operators(op, a, b):
     Outer parametrization is over operators, the inner over param-param, param-number and number-param combinations.
     """
 
-    # Extract underlying numerical value if we are passed a parameter object.
-    try:
-        val_a = a.GetValue()
-    except AttributeError:
-        val_a = a
-    try:
-        val_b = b.GetValue()
-    except AttributeError:
-        val_b = b
+    val_a = to_numeric(a)
+    val_b = to_numeric(b)
 
     assert op(a, b).GetValue() == op(val_a, val_b)
 
 
-@pytest.mark.parametrize('op, a, b', [
-    [ops.pow, const_param(31), 5]])
-def test_incomplete_binary_operators(op, a, b):
+@pytest.mark.parametrize('op, a, b, is_supported', [
```

Comment:
Another option to `is_supported` is to use the `pytest.mark.xfail` decorator. By default, we then expect the test to fail and when it fails the test is reported as `XFAIL`. However, `pytest.mark.xfail` has a `strict` parameter that defines two distinct behaviors for the mark. With `strict=False`, the default, the test will be counted separately as an `XPASS` if it passes. The decorator would simplify the test code to just:

```Python
@pytest.mark.xfail
@pytest.mark.parameterize('op, a, b', [...])
def test_incomplete_binary_operators(op, a, b):
    val_a = to_numeric(a)
    val_b = to_numeric(b)
    assert op(a, b).GetValue() == op(val_a, val_b)
``` 

with output summary (run `pytest -rxX`): 

```
================================================== short test summary info ==================================================
XFAIL ../nest-simulator/testsuite/pytests/test_parameter_operators.py::test_incomplete_binary_operators[pow-a1-b1]
XFAIL ../nest-simulator/testsuite/pytests/test_parameter_operators.py::test_incomplete_binary_operators[pow-31-b2]
XPASS ../nest-simulator/testsuite/pytests/test_parameter_operators.py::test_incomplete_binary_operators[pow-a0-5] 
========================================= 47 passed, 2 xfailed, 1 xpassed in 0.96s ==========================================
```

Note that in this context, both `xfailed` and `xpassed` are regarded as successful tests. 

We might want to consider `pytest.mark.xfail` to document features not yet implemented, as these then can be extracted from the output summary.